FinSearchUnified/Bundle/SearchBundleFindologic/QueryBuilder/NewQueryBuilderFactory.php

```diff
@@ -0,0 +1,267 @@
+<?php
+
+namespace FinSearchUnified\Bundle\SearchBundleFindologic\QueryBuilder;
+
+use Exception;
+use FinSearchUnified\Bundle\SearchBundleFindologic\ConditionHandler\CategoryConditionHandler;
+use FinSearchUnified\Bundle\SearchBundleFindologic\ConditionHandler\PriceConditionHandler;
+use FinSearchUnified\Bundle\SearchBundleFindologic\ConditionHandler\ProductAttributeConditionHandler;
+use FinSearchUnified\Bundle\SearchBundleFindologic\ConditionHandler\SearchTermConditionHandler;
+use FinSearchUnified\Bundle\SearchBundleFindologic\ConditionHandler\SimpleConditionHandler;
+use FinSearchUnified\Bundle\SearchBundleFindologic\ConditionHandlerInterface;
+use FinSearchUnified\Bundle\SearchBundleFindologic\SortingHandler\PopularitySortingHandler;
+use FinSearchUnified\Bundle\SearchBundleFindologic\SortingHandler\PriceSortingHandler;
+use FinSearchUnified\Bundle\SearchBundleFindologic\SortingHandler\ProductNameSortingHandler;
+use FinSearchUnified\Bundle\SearchBundleFindologic\SortingHandler\ReleaseDateSortingHandler;
+use FinSearchUnified\Bundle\SearchBundleFindologic\SortingHandlerInterface;
+use Shopware\Bundle\PluginInstallerBundle\Service\InstallerService;
+use Shopware\Bundle\SearchBundle\ConditionInterface;
+use Shopware\Bundle\SearchBundle\Criteria;
+use Shopware\Bundle\SearchBundle\SortingInterface;
+use Shopware\Bundle\StoreFrontBundle\Struct\ShopContextInterface;
+use Shopware_Components_Config;
+
+class NewQueryBuilderFactory implements NewQueryBuilderFactoryInterface
```

Comment:
Why was `NewQueryBuilderFactoryInterface` created? The interface from the old and the new query builder factory should not change.

The reason for using interfaces is exactly that; to be able to swap out the code behind it without changing the interface. Please implement the interface `QueryBuilderFactoryInterface`.

Reply:
That interface uses the QueryBuilder as the return type. We have implemented the NewQueryBuilder so it is not compatible.

Reply:
I have added that in the description as well in the story.

Reply:
In PHP 5.6 you can not add return types anyway, they are just salt added to some doc block. Changing the docblock should be enough.